providers/azure_private_dns/azurePrivateDnsProvider.go

```diff
@@ -0,0 +1,534 @@
+				rc := &models.RecordConfig{TTL: uint32(*set.Properties.TTL), Original: set}
+				rc.SetLabelFromFQDN(*set.Properties.Fqdn, origin)
+				rc.Type = "TXT"
+				var txts []string
+				for _, txt := range rec.Value {
+					txts = append(txts, *txt)
+				}
+				_ = rc.SetTargetTXTs(txts)
+				results = append(results, rc)
+			}
+		}
+	case "Microsoft.Network/privateDnsZones/MX":
+		for _, rec := range set.Properties.MxRecords {
+			rc := &models.RecordConfig{TTL: uint32(*set.Properties.TTL), Original: set}
+			rc.SetLabelFromFQDN(*set.Properties.Fqdn, origin)
+			rc.Type = "MX"
+			_ = rc.SetTargetMX(uint16(*rec.Preference), *rec.Exchange)
+			results = append(results, rc)
+		}
+	case "Microsoft.Network/privateDnsZones/SRV":
+		for _, rec := range set.Properties.SrvRecords {
+			rc := &models.RecordConfig{TTL: uint32(*set.Properties.TTL), Original: set}
+			rc.SetLabelFromFQDN(*set.Properties.Fqdn, origin)
+			rc.Type = "SRV"
+			_ = rc.SetTargetSRV(uint16(*rec.Priority), uint16(*rec.Weight), uint16(*rec.Port), *rec.Target)
+			results = append(results, rc)
+		}
+	case "Microsoft.Network/privateDnsZones/SOA":
+	default:
+		panic(fmt.Errorf("nativeToRecords rtype %v unimplemented", *set.Type))
+	}
+	return results
+}
+
+// NOTE recordToNativeDiff2 is really "convert []RecordConfig to rrset".
+
+func (a *azurednsProvider) recordToNativeDiff2(recordKey models.RecordKey, recordConfig []*models.RecordConfig) (*adns.RecordSet, adns.RecordType, error) {
+
+	recordKeyType := recordKey.Type
+	//	if recordKeyType == "AZURE_ALIAS" {
+	//		fmt.Fprintf(os.Stderr, "DEBUG: XXXXXXXXXXXXXXXXXXXXXXX %v\n", recordKeyType)
+	//	}
+
+	recordSet := &adns.RecordSet{Type: to.StringPtr(recordKeyType), Properties: &adns.RecordSetProperties{}}
+	for _, rec := range recordConfig {
+		switch recordKeyType {
+		case "A":
+			if recordSet.Properties.ARecords == nil {
+				recordSet.Properties.ARecords = []*adns.ARecord{}
+			}
+			recordSet.Properties.ARecords = append(recordSet.Properties.ARecords, &adns.ARecord{IPv4Address: to.StringPtr(rec.GetTargetField())})
+		case "AAAA":
+			if recordSet.Properties.AaaaRecords == nil {
+				recordSet.Properties.AaaaRecords = []*adns.AaaaRecord{}
+			}
+			recordSet.Properties.AaaaRecords = append(recordSet.Properties.AaaaRecords, &adns.AaaaRecord{IPv6Address: to.StringPtr(rec.GetTargetField())})
+		case "CNAME":
+			recordSet.Properties.CnameRecord = &adns.CnameRecord{Cname: to.StringPtr(rec.GetTargetField())}
+		case "PTR":
+			if recordSet.Properties.PtrRecords == nil {
+				recordSet.Properties.PtrRecords = []*adns.PtrRecord{}
+			}
+			recordSet.Properties.PtrRecords = append(recordSet.Properties.PtrRecords, &adns.PtrRecord{Ptrdname: to.StringPtr(rec.GetTargetField())})
+		case "TXT":
+			if recordSet.Properties.TxtRecords == nil {
+				recordSet.Properties.TxtRecords = []*adns.TxtRecord{}
+			}
+			// Empty TXT record needs to have no value set in it's properties
+			if !(len(rec.TxtStrings) == 1 && rec.TxtStrings[0] == "") {
```

Comment:
A future release is eliminating access to `.TxtStrings` and instead requires the use of `.GetTargetTXTJoined()` or `.GetTargetTXTSegmented()` depending on if you want the whole string or the string segmented into 255-octet chunks.

Let's future-proof this by eliminating access to `.TxtStrings` now.

```suggestion
			if rec.GetTargetTXTJoined() == "" {
```